extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -0,0 +1,267 @@
+     * @param {Object} connectorConfig connector config
+     * @returns {string[]} list of buckets
+     */
+     _extractBucketsFromConfig(connectorConfig) {
+        const pipeline = connectorConfig.pipeline ?
+            JSON.parse(connectorConfig.pipeline) : null;
+        if (!pipeline || pipeline.length === 0) {
+            return [];
+        }
+        return pipeline[0].$match['ns.coll'].$in;
+    }
+
+    /**
+     * Gets old connector configs and initializes connector
+     * instances
+     * @param {string[]} connectorNames connector names
+     * @returns {Promise|Connector[]} list of connectors
+     */
+    async _getOldConnectors(connectorNames) {
+        try {
+            const connectors = Promise.all(connectorNames.map(async connectorName => {
+                // get old connector config
+                const config = await this._kafkaConnect.getConnectorConfig(connectorName);
+                // extract buckets from old connector config and filter them
+                // only leaving currently valid buckets
+                const buckets = this._extractBucketsFromConfig(config);
+                // removing pipeline from config
+                delete config.pipeline;
+                // initializing connector
+                const connector = new Connector({
+                    name: connectorName,
+                    config,
+                    buckets,
+                    logger: this._logger,
+                    kafkaConnectHost: this._kafkaConnectHost,
+                    kafkaConnectPort: this._kafkaConnectPort,
+                });
+                return connector;
+            }));
+            return connectors;
+        } catch (err) {
+            this._logger.error('An error occured while getting old connectors', {
+                method: 'ConnectorsManager._getOldConnectors',
+                error: err.description || err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.description);
+        }
+    }
+
+    /**
+     * Initialize previously created connector instances and
+     * creates new connectors based on configuration
+     * @returns {Promise|Connector[]} list connectors
+     * @throws {InternalError}
+     */
+    async initializeConnectors() {
+        try {
+            // get and initialize old connectors
+            const oldConnectorNames = await this._kafkaConnect.getConnectors();
+            if (oldConnectorNames) {
+                const oldConnectors = await this._getOldConnectors(oldConnectorNames);
+                this._connectors.push(...oldConnectors);
+                this._oldConnectors.push(...oldConnectors);
+            }
+            // Add connectors if required number of connectors not reached
+            const nbConnectorsToAdd = this._nbConnectors - this._connectors.length;
+            if (nbConnectorsToAdd > 0) {
+                for (let i = 0; i < nbConnectorsToAdd; i++) {
```

Comment:
I tend to prefer using `async` module functions such as `async.timesSeries`, it's more explicit and more flexible on what kind of concurrency we're getting, but I'm fine with how it is as a loop.